sdk/include/platform/sunburst/platform-i2c.hh

```diff
@@ -0,0 +1,385 @@
+	  StatusFormatFull = 1 << 0,
+	  /// Host mode Receive FIFO is full
+	  StatusReceiveFull = 1 << 1,
+	  /// Host mode Format FIFO is empty
+	  StatusFormatEmpty = 1 << 2,
+	  /// Host functionality is idle. No Host transaction is in progress
+	  StatusHostIdle = 1 << 3,
+	  /// Target functionality is idle. No Target transaction is in progress
+	  StatusTargetIdle = 1 << 4,
+	  /// Host mode Receive FIFO is empty
+	  SmatusReceiveEmpty = 1 << 5,
+	  /// Target mode Transmit FIFO is full
+	  StatusTransmitFull = 1 << 6,
+	  /// Target mode Receive FIFO is full
+	  StatusAcquiredFull = 1 << 7,
+	  /// Target mode Transmit FIFO is empty
+	  StatusTransmitEmpty = 1 << 8,
+	  /// Target mode Aquired FIFO is empty
+	  StatusAcquiredEmpty = 1 << 9,
+	  /**
+	   * A Host-Mode active transaction has been ended by the
+	   * HostNackHandlerTimeout mechanism. This bit is cleared when
+	   * Control.EnableHost is set by software to start a new transaction.
+	   */
+	  StatusHostDisabledNackTimeout = 1 << 10,
+	};
+
+	/// FormatData Register Fields
+	enum [[clang::flag_enum]] : uint32_t{
+	  /// Issue a START condition before transmitting BYTE.
+	  FormatDataStart = 1 << 8,
+	  /// Issue a STOP condition after this operation
+	  FormatDataStop = 1 << 9,
+	  /// Read BYTE bytes from I2C. (256 if BYTE==0)
+	  FormatDataReadBytes = 1 << 10,
+	  /**
+	   * Do not NACK the last byte read, let the read
+	   * operation continue
+	   */
+	  FormatDataReadCount = 1 << 11,
+	  /// Do not signal an exception if the current byte is not ACK’d
+	  FormatDataNakOk = 1 << 12,
+	};
+
+	/// FifoControl Register Fields
+	enum [[clang::flag_enum]] : uint32_t{
+	  /// Receive fifo reset. Write 1 to the register resets it. Read returns 0
+	  FifoControlReceiveReset = 1 << 0,
+	  /// Format fifo reset. Write 1 to the register resets it. Read returns 0
+	  FifoControlFormatReset = 1 << 1,
+	  /// Aquired FIFO reset. Write 1 to the register resets it. Read returns 0
+	  FifoControlAcquiredReset = 1 << 7,
+	  /// Transmit FIFO reset. Write 1 to the register resets it. Read returns 0
+	  FifoControlTransmitReset = 1 << 8,
+	};
+
+	/**
+	 * Specification times (Table 10) in nanoseconds for each bus mode.
+	 */
+	static constexpr uint16_t SpcThigh[]  = {4000u, 600u, 260u};
+	static constexpr uint16_t SpcTlow[]   = {4700u, 1300u, 150u};
+	static constexpr uint16_t SpcThdSta[] = {4000u, 600u, 260u};
+	static constexpr uint16_t SpcTsuSta[] = {4700u, 600u, 260u};
+	static constexpr uint16_t SpcThdDat[] = {4000u, 1u, 1u};
+	static constexpr uint16_t SpcTsuDat[] = {500u, 100u, 50u};
+	static constexpr uint16_t SpcTBuf[]   = {4700u, 1300u, 500u};
+	static constexpr uint16_t SpcTsuSto[] = {4000u, 600u, 260u};
+
+	/// Reset all of the fifos
+	void reset_fifos() volatile
+	{
+		fifoCtrl = (FifoControlReceiveReset | FifoControlFormatReset |
+		            FifoControlAcquiredReset | FifoControlTransmitReset);
+	}
+
+	/// Configure the I2C block to be in host mode
+	void set_host_mode() volatile
```

Comment:
We normally name things noun-verb, so that similar things are sorted together when autocompleting or reading generated docs.  Ideally avoid starting things with `set` because it ends up with a load of unrelated `set_` things in autocomplete.

My preferred style here (which we haven't done uniformly) is to have two overloads with the name of the thing, one of which takes an argument and updates it, the other that takes no arguments and does not update it.